src/ImageSharp/Processing/Extensions/Effects/PixelRowDelegateExtensions.cs

```diff
@@ -55,14 +155,114 @@ public static IImageProcessingContext ProcessPixelRowsAsVector4(this IImageProce
         public static IImageProcessingContext ProcessPixelRowsAsVector4(this IImageProcessingContext source, PixelRowOperation rowOperation, Rectangle rectangle, PixelConversionModifiers modifiers)
             => source.ApplyProcessor(new PixelRowDelegateProcessor(rowOperation, modifiers), rectangle);
 
+        /// <summary>
+        /// Applies a user defined processing delegate to the image.
+        /// </summary>
+        /// <typeparam name="TDelegate">The type of user defined, row processing delegate to use.</typeparam>
+        /// <param name="source">The image this method extends.</param>
+        /// <returns>The <see cref="IImageProcessingContext"/> to allow chaining of operations.</returns>
+        public static IImageProcessingContext ProcessPositionAwarePixelRowsAsVector4<TDelegate>(this IImageProcessingContext source)
```

Comment:
I don't like the additional naming here. It was fine as it was with `ProcessPixelRowsAsVector4`. The generic variant was visible enough

Reply:
The reason why I did that is that you can't have overloads that are only different for the type constraints. Here we'd have two methods with the same name and parameter, with the only difference being that one takes a `TDelegate where TDelegate : struct, IPixelRowDelegate`, and the other takes a `TDelegate where TDelegate : struct, IPixelRowDelegate<Point>`.

Not sure how to avoid having the separate name, unless we dropped the value delegate API entirely for the position aware variant, but we'd then lose half the new APIs basically, and also make the API surface "asymmetrical" in this case, if that matters 🤔